vmms/dockerSSH.py

```diff
@@ -0,0 +1,231 @@
+    """ 
+
+    # Launch the command
+    p = subprocess.Popen(command,
+                        stdout=open("/dev/null", 'w'),
+                        stderr=subprocess.STDOUT)
+
+    # Wait for the command to complete
+    t = 0.0
+    while t < time_out and p.poll() is None:
+        time.sleep(config.Config.TIMER_POLL_INTERVAL)
+        t += config.Config.TIMER_POLL_INTERVAL
+
+    # Determine why the while loop terminated
+    if p.poll() is None:
+        subprocess.call(["/bin/kill", "-9", str(p.pid)])
+        returncode = -1
+    else:
+        returncode = p.poll()
+    return returncode
+
+def timeoutWithReturnStatus(command, time_out, returnValue = 0):
+    """ timeoutWithReturnStatus - Run a Unix command with a timeout,
+    until the expected value is returned by the command; On timeout,
+    return last error code obtained from the command.
+    """
+    p = subprocess.Popen(command, 
+                        stdout=open("/dev/null", 'w'), 
+                        stderr=subprocess.STDOUT)
+    t = 0.0
+    while (t < time_out):
+        ret = p.poll()
+        if ret is None:
+            time.sleep(config.Config.TIMER_POLL_INTERVAL)
+            t += config.Config.TIMER_POLL_INTERVAL
+        elif ret == returnValue:
+            return ret
+        else:
+            p = subprocess.Popen(command,
+                            stdout=open("/dev/null", 'w'),
+                            stderr=subprocess.STDOUT)
+    return ret
+
+def dockerExec(container, cmd, time_out=1):
+    """ docerExec - Executes `docker exec container cmd` and
+        returns output. Container is the name of the docker
+        container and cmd is a list of commands to run.
+    """
+    command = ['docker', 'exec', container, 'sh', '-c'] + cmd
+    return timeout(command, time_out)
+
+#
+# User defined exceptions
+#
+
+class DockerSSH:
+    _OS_X = 'darwin'
+    LOCALHOST = '127.0.0.1'
+
+    def __init__(self):
+        """
+			Checks if the machine is ready to run docker containers.
+            Initialize boot2docker if running on OS X.
+        """
+        try:
+            self.log = logging.getLogger("DockerSSH")
+            # If running on OS X, create a boot2docker VM
+            if sys.platform == self._OS_X:
+                # boot2docker initialization will be part of initial
+                # set up with Tango.
+                self.docker_host_ip = subprocess.check_output(['boot2docker', 'ip']).strip('\n')
+            else:
+                self.docker_host_ip = self.LOCALHOST
+
+            self.log.info("Docker host IP is %s" % self.docker_host_ip)
+
+        except Exception as e:
+            self.log.error(str(e))
+            exit(1)
+
+    def instanceName(self, id, name):
+        """ instanceName - Constructs a VM instance name. Always use
+        this function when you need a VM instance name. Never generate
+        instance names manually.
+        """
+        # return str(id)
```

Comment:
Something was inconsistent about the instance names so I was trying out different things. Gosh, fine, I'll get rid of it!